tests/unit/test_cli.py

```diff
@@ -122,6 +122,23 @@ def test_get_all(cli_runner, vault_with_token):
     assert result.exit_code == 0
 
 
+def test_get_all_flat(cli_runner, vault_with_token):
+
+    vault_with_token.db = {
+        "a/baz": {"value": "bar"},
+        "a/foo": {"value": ["y", "ay"]},
+        "a/db": {"host": "host.name", "port": 5432},
+    }
+    result = cli_runner.invoke(cli.cli, ["get-all", "--flat", "a"])
+
+    assert yaml.safe_load(result.output) == {
+        "a/baz": "bar",
+        "a/foo": ["y", "ay"],
+        "a/db": {"host": "host.name", "port": 5432},
```

Comment:
It feels like this last line is testing something else entirely but ok :) I see where you want this to go, and even if this is already tested elsewhere, it's ok. But be careful not to overcomplicate tests, it makes them harder to maintain :) If there's the slightest chance this line catches something another test wouldn't have caught, it's worth it :+1:

Reply:
when we will remove the `value` default key, only this last case will make sense :)

Reply:
Given what this function does, are we testing something different in the 3 lines ?

Reply:
no.
do you want me to fix that ?

Reply:
I don't know. As I said, I'm ok to keep it, but I just think we have to be mindful of this.